### ISSUE TITLE
Nesting counts enum cases wrong

### ISSUE THREAD

**Issue by juri** [realm/SwiftLint#2357]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

It looks like the nesting rule counts cases of deeply nested enum with the statement rule while ignoring the nesting level of the enum which is in acceptable bounds. So specify `type_level` as something well above the statement level, build a pyramid of ~~doom~~ nested types with the enum as the deepest type, and you'll get a warning about the level of nesting of the cases. Example below.

This issue seemed to be related to #190, but I guess is not quite the same?

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint

Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'Nest.swift' (1/1)
/tmp/foo/Nest.swift:6:26: warning: Nesting Violation: Statements should be nested at most 5 levels deep (nesting)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.27.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
nesting:
    type_level: 4
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcode-select -p`)? 9.4.1
* Example:

```swift
struct Foo {
    struct Bar {
        struct Zap {
            struct Blarp {
                enum Target: Equatable {
                    case contentSourceParam(name: String)
                }
            }
        }
    }
}
```
